### ISSUE TITLE
2.6.2 instantly crashes at start 

### ISSUE THREAD

**Issue by gunwald** [nextcloud/desktop#1704]
### Expected behaviour
The desktop client should start up ans run.

### Actual behaviour
The desktop Linux client crashes on start up with a segmentation fault:
```
~$ nextcloud
Gtk-Message: 11:29:46.893: Failed to load module "$GTK3_MODULES"
Gtk-Message: 11:29:46.893: Failed to load module "/usr/lib/x86_64-linux-gnu/libplotinus/libplotinus.so"
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
Fontconfig warning: "/etc/fonts/conf.avail/29-language-selector-ja-jp.conf", line 55: Having multiple values in <test> isn't supported and may not work as expected
Fontconfig warning: "/etc/fonts/conf.avail/29-language-selector-zh.conf", line 8: Having multiple values in <test> isn't supported and may not work as expected

** (nextcloud:9212): CRITICAL **: 11:29:47.907: cloud_providers_dbus_object_skeleton_new: assertion 'g_variant_is_object_path (object_path)' failed

(nextcloud:9212): GLib-GObject-CRITICAL **: 11:29:47.907: g_object_set: assertion 'G_IS_OBJECT (object)' failed
Speicherzugriffsfehler (Speicherabzug geschrieben)
```

### Steps to reproduce
1. Install the actual version 2.6.2 of the Nextcloud client on an Debian based system.
2. Start the client.

### Client configuration
Client version:  2.6.2 

Operating system: Ubuntu 19.10

OS language: german

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only): Nextcloud ppa

Installation path of client:
```
~$ whereis nextcloud
nextcloud: /usr/bin/nextcloud /usr/lib/x86_64-linux-gnu/nextcloud /usr/share/nextcloud
```

**Comment by jurgenhaas** [comment]
I'm seeing the exact same issue on Ubuntu 18.04 LTS.

**Comment by menemat** [comment]
Same here.
When I try to expand the folder structure client crash.
Windows 10, client version 2.6.2

**Comment by krfgbhoiaeuzr** [comment]
I seem to have the same problem, fresh client install just crashes silently when i try to open it.
Windows 10, client version 2.6.2

**Comment by silverspr** [comment]
I am having the same problem on Ubuntu 18.04, crashes after install, appears briefly in panel then disappears.  Tried uninstalling and reinstalling, same behaviour.  Most silent crash.

**Comment by pgassmann** [comment]
Same here. immediately crashes since update.

```
$ nextcloud

** (nextcloud:26326): CRITICAL **: 17:23:32.138: cloud_providers_dbus_object_skeleton_new: assertion 'g_variant_is_object_path (object_path)' failed

(nextcloud:26326): GLib-GObject-CRITICAL **: 17:23:32.138: g_object_set: assertion 'G_IS_OBJECT (object)' failed
Speicherzugriffsfehler (Speicherabzug geschrieben)
```

**Comment by pgassmann** [comment]
@misch7 Can you investigate this? This is important! Thank you! Tell me if I can help with more information. `strace` output (fragment below) is not clear to me.
@ivaradi you uploaded the package to the ppa. where can I find the older versions? is this a restriction of the launchpad ppa to have only the latest release available?

```
futex(0x7f04f4a91f38, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x7f04f4a91f38, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x7f04f4a91f38, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x7f04f4a91f38, FUTEX_WAKE_PRIVATE, 2147483647) = 0
write(11, "\1\0\0\0\0\0\0\0", 8)        = 8
futex(0x7f04bc01b330, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc01b0c0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc013018, FUTEX_WAKE_PRIVATE, 1) = 1
write(11, "\1\0\0\0\0\0\0\0", 8)        = 8
futex(0x7f04bc01b330, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc01b0c0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc013018, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04f43dc288, FUTEX_WAIT_PRIVATE, 2, NULL) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x7f04f43dc288, FUTEX_WAKE_PRIVATE, 1) = 0
write(11, "\1\0\0\0\0\0\0\0", 8)        = 8
futex(0x7f04bc01b330, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc01b0c0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f04bc013018, FUTEX_WAKE_PRIVATE, 1) = 1
getpeername(2, 0x7ffd13d39140, [128])   = -1 ENOTSOCK (Socket operation on non-socket)
futex(0x7f04f4a91f38, FUTEX_WAKE_PRIVATE, 2147483647) = 0
ioctl(2, TCGETS, {B38400 opost isig icanon echo ...}) = 0
getpid()                                = 26756
stat("/etc/localtime", {st_mode=S_IFREG|0644, st_size=1918, ...}) = 0
write(2, "\n** (nextcloud:26756): \33[1;35mCR"..., 170
** (nextcloud:26756): CRITICAL **: 17:27:46.591: cloud_providers_dbus_object_skeleton_new: assertion 'g_variant_is_object_path (object_path)' failed
) = 170
ioctl(2, TCGETS, {B38400 opost isig icanon echo ...}) = 0
getpid()                                = 26756
stat("/etc/localtime", {st_mode=S_IFREG|0644, st_size=1918, ...}) = 0
write(2, "\n(nextcloud:26756): GLib-GObject"..., 134
(nextcloud:26756): GLib-GObject-CRITICAL **: 17:27:46.592: g_object_set: assertion 'G_IS_OBJECT (object)' failed
) = 134
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=NULL} ---
+++ killed by SIGSEGV (core dumped) +++
Speicherzugriffsfehler (Speicherabzug geschrieben)
```

**Comment by pgassmann** [comment]
**Update** @ivaradi @misch7 I can start the latest `./Nextcloud-2.6.2-x86_64.AppImage`, but `nextcloud` installed from the ppa.

**Comment by AlbrechtL** [comment]
Same for me Ubuntu 19.10. I'm using version `2.6.2-20191224.111326~eoan1 `from ppa. Version `2.6.1-20191017.124538~eoan1` works fine.

**Comment by smatthie** [comment]
Same issue here, segmentation fault with same error message. This started just yesterday with an update to 2.6.2 and was working perfectly fine until then.

I have one directory with an Umlaut (Arbeitsfläche - German for desktop), when I remove the entries for this directory from the nextcloud.cfg file, it starts up correctly (but of course doesn't sync that directory any more).

**Comment by AlbrechtL** [comment]
> Same for me Ubuntu 19.10.

I'm also German and we definitely have the characters ö,ä,ü somewhere. In my case nextcloud-client starts but when it is starting sycing it crashes.

**Comment by smatthie** [comment]
Could somebody please look at this? I don't want to create pressure, but I haven't been able to start nextcloud and synchronise for 2 weeks now. Thanks!

**Comment by smatthie** [comment]
I can start the AppImage, but not the PPA version.

**Comment by ivaradi** [comment]
@tintou Could you look at this, please? 

It works for me correctly, but seemingly the call to `cloud_providers_account_exporter_new()` crashes in the constructor of `CloudProviderWrapper`. I managed to reproduce this error in a small program by passing string containing accented characters as the second argument to the function. But I cannot see how `accountName` could contain any such characters. (BTW converting to UTF-8 is not really meaningful there, as `g_variant_is_object_path()` accepts only English letters, digits and underscore: https://developer.gnome.org/glib/stable/glib-GVariant.html#g-variant-is-object-path).

Could such characters come from an old configuration file read by `FolderMan::setupFolderFromOldConfigFile`?

**Comment by smatthie** [comment]
Thanks for looking at this. As you mention the config file, here is my nextcloud.cfg (redacted to remove certificates and some personal info). There are some entries that look very weird. 
I had owncloud for many years and migrated to nextcloud a few years ago, so there might well be some old stuff causing problems. 

[nextcloud.cfg.txt](https://github.com/nextcloud/desktop/files/4133570/nextcloud.cfg.txt)

**Comment by ivaradi** [comment]
@smatthe thanks for the file. In the `[Accounts]` section there are the `0\Folders\XXXX\....=...` entries. Could you try modifying the  `XXXX` parts so that each different `XXXX` is replaced with a different number. Say, `Arbeitsflache` becomes 2, `Bilder` becomes 3, and so on.

**Comment by smatthie** [comment]
@ivaradi Yes, with this change nextcloud starts and synchronises normally. So the entries now look like this:

> 0\Folders\2\ignoreHiddenFiles=false
> 0\Folders\2\journalPath=._sync_1384f5ecf9ef.db
> 0\Folders\2\localPath=/home/stephan/Arbeitsfl\xe4\x63he/
> 0\Folders\2\paused=false
> 0\Folders\2\targetPath=/Arbeitsfl\xe4\x63he

**Comment by ivaradi** [comment]
@smatthie thanks, so we know what the problem is, and there is a workaround at least. 

I am not sure what the real solution could be, but hopefully @tintou will look at it :)

That said, the reason why the AppImage client works but the PPA one does not is that the PPA version is linked with the `libcloudproviders` library, while the AppImage one is not.

**Comment by smatthie** [comment]
@ivaradi Thank you very much for your help!

**Comment by tintou** [comment]
Thanks for the debugging, I've found the issue which PR #1763 fixes

**Comment by pgassmann** [comment]
Thank you @ivaradi @smatthie @tintou 
I could narrow down the issue to two folders that are locally on an external disk. The external disk was not attached for some months now and the synchronisation of these two folders were paused. 
But it does not depend on whether the disk is mounted or not, the nextcloud client starts without issue when running `Nextcloud-2.6.2-x86_64.AppImage`, but it crashes immediately when starting `nextcloud` from the ppa.
I tried also deleting the .sync files and syncing the folders again with the AppImage. `nextcloud` still crashes also now that the disk is mounted and the folders are synced.

@tintou is this the issue which is fixed in #1763 ?
@ivaradi can you explain the difference in behaviour between the two packaging methods?

```
$ nextcloud --version
Nextcloud version 2.6.2git
Using Qt 5.9.5, built against Qt 5.9.5
Using 'OpenSSL 1.1.1  11 Sep 2018'
$ Nextcloud-2.6.2-x86_64.AppImage --version
Nextcloud version 2.6.2stable (build 20191224)
Git revision 1d745535f7a95ff2265d6b6b47121200338fce83
Using Qt 5.12.5, built against Qt 5.12.5
Using 'OpenSSL 1.1.1d  10 Sep 2019'
```

Output when crashing
```
$ nextcloud

** (nextcloud:15922): CRITICAL **: 13:14:00.766: cloud_providers_dbus_object_skeleton_new: assertion 'g_variant_is_object_path (object_path)' failed

(nextcloud:15922): GLib-GObject-CRITICAL **: 13:14:00.766: g_object_set: assertion 'G_IS_OBJECT (object)' failed
Speicherzugriffsfehler (Speicherabzug geschrieben)
```

The two folders in `nextcloud.cfg`:
```
0\Folders\Website-Media\ignoreHiddenFiles=true
0\Folders\Website-Media\journalPath=._sync_a81ee788cbec.db
0\Folders\Website-Media\localPath=/media/pgassmann/datadisk/BGF/Webseite/Website-Media/
0\Folders\Website-Media\paused=false
0\Folders\Website-Media\targetPath=/Website-Media
0\Folders\download\ignoreHiddenFiles=true
0\Folders\download\journalPath=._sync_94ba7c8b6bbd.db
0\Folders\download\localPath=/media/pgassmann/datadisk/BGF/Webseite/download/
0\Folders\download\paused=false
0\Folders\download\targetPath=/Website-Downloads/download
```

**Comment by tintou** [comment]
Yes this issue is fixed by #1763
